core/test/histogram_test.cpp

```diff
@@ -45,38 +45,49 @@ auto make_single_sparse() {
   return sparse;
 }
 
+DataArray make_expected(const Variable &var, const Variable &edges) {
+  auto dim = var.dims().inner();
+  std::map<Dim, Variable> coords = {{dim, edges}};
+  auto expected = DataArray(var, coords, std::map<std::string, Variable>());
+  return expected;
+}
+
 TEST(HistogramTest, below) {
   const auto sparse = make_single_sparse();
   auto edges = makeVariable<double>({Dim::X, 3}, {-2.0, -1.0, 0.0});
   auto hist = core::histogram(sparse[""], edges);
-  EXPECT_EQ(hist,
-            makeVariable<double>({Dim::X, 2}, units::counts, {0, 0}, {0, 0}));
+  std::map<Dim, Variable> coords = {{Dim::X, edges}};
+  auto expected = make_expected(
+      makeVariable<double>({Dim::X, 2}, units::counts, {0, 0}, {0, 0}), edges);
+  EXPECT_EQ(hist, expected);
```

Comment:
I think it would make sense to preserve the name of the input?

Reply:
I'm not sure I understand what you mean..

Reply:
The input is `DataArray`, which has a name, but the output defaults to `""`?

Reply:
But doesn't a `DataArray` have a name only when it is part of a `Dataset`? so what `histogram` returns can never have a name?

Reply:
No, `DataArray` has a name. The name of `DataProxy` is the name of the item in the dataset.

Reply:
So does the same go for your implementation of `rebin`? (https://github.com/scipp/scipp/blob/5abee323622330e8b1dff95c45c6da6db1340dc7/core/dataset.cpp#L1203)
We should also keep the name in the `DataArray` that is being returned?

Reply:
Good point.

In general, is there any reason for *not* keeping the name?